examples/gke/deploy-paligemma-2-with-tgi/README.md

```diff
@@ -0,0 +1,403 @@
+
+The PaliGemma and PaliGemma2 papers use the BOS token after the images and before the prefix and then `\n` i.e. the line-break, as the separator token from suffix (input) and the prefix (output); which are both automatically included by the `transformers.PaliGemmaProcessor`, meaning that there's no need to provide those explicitly to the `/generate` endpoint in TGI.
+
+Besides that, the images should be provided following the Markdown formatting for image rendering i.e. `![](<image_url>)`, and the image needs to be publicly accessible; or provided as its base64 encoding if not hosted within a publicly accessible URL.
+
+This means that the prompt formatting expected on the `/generate` method is either:
+
+- `![](<URL>)<PROMPT>` if the image is provided via URL.
+- `![](data:image/png;base64,<ENCODING>)<PROMPT>` if the image is provided as its base64 encoding.
+
+Read more information about the technical details and implementation of PaliGemma on the papers / technical reports released by Google:
+
+- [PaliGemma: A versatile 3B VLM for transfer](https://arxiv.org/abs/2407.07726)
+- [PaliGemma 2: A Family of Versatile VLMs for Transfer](https://arxiv.org/abs/2412.03555)
+
+> [!NOTE]
+> Note that the `/v1/chat/completions` endpoint cannot be used, and will result in a "chat template error not found", as the model is pre-trained and not fine-tuned for e.g. chat conversations, and does not have a chat template defined that can be applied within the `v1/chat/completions` endpoint following the OpenAI OpenAPI specification.
+
+### Via cURL
+
+To send a POST request to the TGI service using `cURL`, you can run the following command:
+
+```bash
+curl http://localhost:8080/generate \
+    -d '{"inputs":"![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/rabbit.png)caption en","parameters":{"max_new_tokens":128,"seed":42}}' \
+    -H 'Content-Type: application/json'
+```
+
+| Image                                                                                                      | Input      | Output                        |
+|------------------------------------------------------------------------------------------------------------|------------|-------------------------------|
+| ![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/rabbit.png) | caption en | image of a man in a spacesuit |
+
+### Via Python
+
+You can install it via pip as `pip install --upgrade --quiet huggingface_hub`, and then run the following snippet to mimic the cURL command above i.e. sending requests to the Generate API:
+
+```python
+from huggingface_hub import InferenceClient
+
+client = InferenceClient("http://localhost:8080", api_key="-")
+
+generation = client.text_generation(
+    prompt="![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/rabbit.png)caption en",
```

Comment:
we are missing newline here

Reply:
Indeed after digging the `text-generation-inference` code I realised that's automatically injected, see https://github.com/huggingface/text-generation-inference/blob/bb9095aae339579fbf3b4e7be3909932de26a7ee/server/text_generation_server/models/pali_gemma.py#L31